tests/README.md

```diff
@@ -0,0 +1,173 @@
+# Testing Guide
+
+We have a variety of tests implemented as a support to Agent Factory development.
+This guide describes them in detail and explains how to run them locally.
+The following sections are:
+
+- **Directory Structure**, explaining what is available and where.
+- **Running Tests**, describing how to run each different type of test.
+- **Tests in CI**, showing how tests are run on GitHub CI.
+
+## Directory Structure
+
+The `tests` folder is organized in the following sub-directories:
+
+- **`unit`**: unit tests for core functionality including e.g. a2a client/server,
+agent generation and validation, io and storage.
+- **`tools`**: tests for MCP server discovery and our built-in MCP tools.
```

Comment:
This is not that important, but why do we keep the tests for `tools` in a different directory? Aren't they just unit tests?